systemd/check-cloud-env.sh

```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+if grep -qi 'ec2' /sys/class/dmi/id/product_uuid ||
+   grep -qi 'google' /sys/class/dmi/id/product_name ||
+   grep -qi 'microsoft' /sys/class/dmi/id/sys_vendor; then
+    exit 0
+else
+    exit 1
+fi
```

Comment:
Not asking for a change/retest/…, but I believe the `if/else/exit` is not required, this would return the same exit code:
```
grep -qi 'ec2' /sys/class/dmi/id/product_uuid ||
   grep -qi 'google' /sys/class/dmi/id/product_name ||
   grep -qi 'microsoft' /sys/class/dmi/id/sys_vendor
```
But this would most likely need a comment to make it clear to readers which exit codes are expected :) So it’s good as it is, more explicit.